SimTracker/SiPhase2Digitizer/plugins/PixelDigitizerAlgorithm.cc

```diff
@@ -240,3 +246,67 @@ bool PixelDigitizerAlgorithm::isAboveThreshold(const DigitizerUtility::SimHitInf
   } else
     return true;
 }
+//
+// -- Read Bad Channels from the Condidion DB and kill channels/module accordingly
+//
+void PixelDigitizerAlgorithm::module_killing_DB(const Phase2TrackerGeomDetUnit* pixdet) {
+  bool isbad = false;
+  uint32_t detID = pixdet->geographicalId().rawId();
+  int ncol = pixdet->specificTopology().ncolumns();
+  if (ncol < 0)
+    return;
+  std::vector<SiPixelQuality::disabledModuleType> disabledModules = siPixelBadModule_->getBadComponentList();
+
+  SiPixelQuality::disabledModuleType badmodule;
+  for (size_t id = 0; id < disabledModules.size(); id++) {
+    if (detID == disabledModules[id].DetID) {
+      isbad = true;
+      badmodule = disabledModules[id];
+      break;
+    }
+  }
+
+  if (!isbad)
+    return;
+
+  signal_map_type& theSignal = _signal[detID];  // check validity
+  if (badmodule.errorType == 0) {               // this is a whole dead module.
+    for (auto& s : theSignal)
+      s.second.set(0.);  // reset amplitude
+  } else {               // all other module types: half-modules and single ROCs.
+    // Get Bad ROC position:
+    // follow the example of getBadRocPositions in CondFormats/SiPixelObjects/src/SiPixelQuality.cc
+    std::vector<GlobalPixel> badrocpositions;
+    for (size_t j = 0; j < static_cast<size_t>(ncol); j++) {
+      if (siPixelBadModule_->IsRocBad(detID, j)) {
+        std::vector<CablingPathToDetUnit> path = fedCablingMap_->pathToDetUnit(detID);
+        for (auto const& p : path) {
+          const PixelROC* myroc = fedCablingMap_->findItem(p);
+          if (myroc->idInDetUnit() == j) {
+            LocalPixel::RocRowCol local = {39, 25};  //corresponding to center of ROC row, col
```

Comment:
I do not think so. This is basically from PhaseI. Should I make it dummy for the moment? As it is the case fpr PSP sensors?

Reply:
As the functionality is not used,  I think it's fine as it is. Let's revisit this after the PR is merged.

Reply:
These hardcoded bad modules should better get taken from a DB or configuration file.
It is easy to forget them hardcoded here otherwise.
If the experts keep agreeing on merging this PR as such, we will mergi it as it is: please at least open a github issue in that case, so that it will remain track of it.

Reply:
@perrotta this are NOT hardcoded bad modules, this is just a cheap way of getting the center of the ROC in local pixel coordinates. My point was that 39,25 is the middle of the ROC for the phase-1 pixel, but not necessarily also for the phase-2 one.

Reply:
Ah, ok. As such it makes more sense.
In any case, this could also get made customizable via python config, if it has to eventually support a different geometry for the Phase2 modules.